wdl-lint/src/rules/malformed_lint_directive.rs

```diff
@@ -0,0 +1,139 @@
+//! A lint rule for flagging malformed lint directives.
+
+use std::collections::HashMap;
+use std::sync::LazyLock;
+use wdl_ast::AstToken;
+use wdl_ast::Comment;
+use wdl_ast::EXCEPT_COMMENT_PREFIX;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::SyntaxElement;
+use wdl_ast::SyntaxKind;
+use wdl_ast::ToSpan;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+use crate::rules;
+use crate::Rule;
+use crate::rules::RULE_MAP;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the Malformed Lint Directive rule.
+const ID: &str = "MalformedLintDirective";
+
+/// Creates a "Malformed Lint Directive" diagnostic.
+fn malformed_lint_directive(
+    id: &str,
+    span: Span,
+    wrong_element: &SyntaxElement,
+    exceptable_nodes: &[SyntaxKind],
+) -> Diagnostic {
+    let locations = exceptable_nodes
+        .iter()
+        .map(|node| node.describe())
+        .collect::<Vec<_>>()
+        .join(", ");
+
+    Diagnostic::warning(format!(
+        "lint directive `{id}` is malformed"
+    ))
+        .with_label("cannot make an exception for this rule", span)
+        .with_label(
+            "malformed lint directive",
+            wrong_element.text_range().to_span(),
+        )
+        .with_fix(format!(
+            "format your lint direct with as #@ except: {locations}"
+        ))
+}
+
+/// Detects a malformed lint directive.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct MalformedLintDirectiveRule;
+
+impl Rule for MalformedLintDirectiveRule {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Flags malformed lint directives"
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Comments which begin with `#@` must only contain valid lint directives. Lint directives \
+        must be on their own line, only preceded by whitespace."
+    }
+
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Clarity, Tag::Correctness])
+    }
+
+    fn exceptable_nodes(&self) -> Option<&'static [wdl_ast::SyntaxKind]> {
+        // Some(&[SyntaxKind::VersionStatementNode])
+        None
+    }
+}
+
+impl Visitor for MalformedLintDirectiveRule {
+    type State = Diagnostics;
+
+    fn document(&mut self, _: &mut Self::State, _: VisitReason, _: &Document, _: SupportedVersion) {
+        // This is intentionally empty, as this rule has no state.
+    }
+
+    fn comment(&mut self, state: &mut Self::State, comment: &Comment) {
+        // println!("Comment: {:?}", comment);
+        if let x = comment.syntax().prev_token() {
+            let y = x.unwrap().kind();
+            println!("{:?}", y);
+        }
```

Comment:
For production code, we generally prefer using `expect("<the assumption we're making>")` instead of `unwrap()`. This is because `expect()` is self-documenting, where the message communicates the assumption. Whereas `unwrap()` is usually accompanied by a comment explaining the assumption. `expect()` is just a bit nicer than `unwrap()`, but they accomplish the same thing.

But the key for both of them is that we only use them when we are dang-near positive they won't `panic` in production. Anytime they get hit, that's considered an unhandled bug (IMO at least).

Here, a comment may or may not have a `prev_token()` so the `None` case needs to be handled. You probably want something like:
```suggestion
        if let Some(prev) = comment.syntax().prev_token() {
            // handle the case that something occurs before the comment
        } else {
            // handle the case that the comment is the first thing in the document
        };
```